includes/class-amp-theme-support.php

```diff
@@ -303,13 +337,246 @@ public static function is_customize_preview_iframe() {
+			if ( AMP_Post_Meta_Box::DISABLED_STATUS === get_post_meta( get_option( 'page_for_posts' ), AMP_Post_Meta_Box::STATUS_POST_META_KEY, true ) ) {
+				/* translators: %s is the URL to the edit post screen */
+				$templates['is_home']['description'] = sprintf( __( 'Currently disabled at the <a href="%s" target="_blank">page level</a>.', 'amp' ), esc_url( get_edit_post_link( get_option( 'page_for_posts' ) ) ) );
+			}
+		} else {
+			$templates['is_home'] = array(
+				'label' => __( 'Homepage', 'amp' ),
+			);
+		}
+
+		$templates = array_merge(
+			$templates,
+			array(
+				'is_archive' => array(
+					'label' => __( 'Archives', 'amp' ),
+				),
+				'is_author'  => array(
+					'label'  => __( 'Author', 'amp' ),
+					'parent' => 'is_archive',
+				),
+				'is_date'    => array(
+					'label'  => __( 'Date', 'amp' ),
+					'parent' => 'is_archive',
+				),
+				'is_search'  => array(
+					'label' => __( 'Search', 'amp' ),
+				),
+				'is_404'     => array(
+					'label' => __( 'Not Found (404)', 'amp' ),
+				),
+//				'other'    => array(
+//					'label'    => __( 'Other', 'amp' ),
+//					'callback' => '__return_true', // @todo This can be treated as the parent of all, as the default.
+//				),
+			)
+		);
+
+		if ( taxonomy_exists( 'category' ) ) {
+			$templates['is_category'] = array(
+				'label'  => get_taxonomy( 'category' )->labels->name,
+				'parent' => 'is_archive',
+			);
+		}
+		if ( taxonomy_exists( 'post_tag' ) ) {
+			$templates['is_tag'] = array(
+				'label'  => get_taxonomy( 'post_tag' )->labels->name,
+				'parent' => 'is_archive',
+			);
+		}
+
+		$taxonomy_args = array(
+			'_builtin'           => false,
+			'publicly_queryable' => true,
+		);
+		foreach ( get_taxonomies( $taxonomy_args, 'objects' ) as $taxonomy ) {
+			$templates[ sprintf( 'is_tax[%s]', $taxonomy->name ) ] = array(
+				'label'    => $taxonomy->labels->name,
+				'parent'   => 'is_archive',
+				'callback' => function ( WP_Query $query ) use ( $taxonomy ) {
+					return $query->is_tax( $taxonomy->name );
+				},
+			);
+		}
+
+		$post_type_args = array(
+			'has_archive'        => true,
+			'publicly_queryable' => true,
+		);
+		foreach ( get_post_types( $post_type_args, 'objects' ) as $post_type ) {
+			$templates[ sprintf( 'is_post_type_archive[%s]', $post_type->name ) ] = array(
+				'label'    => $post_type->labels->archives,
+				'callback' => function ( WP_Query $query ) use ( $post_type ) {
+					return $query->is_post_type_archive( $post_type->name );
+				},
+			);
+		}
+
+		/**
+		 * Filters list of supportable templates.
+		 *
+		 * A theme or plugin can force a given template to be supported or not by preemptively
+		 * setting the 'supported' flag for a given template. Otherwise, if the flag is undefined
+		 * then the user will be able to toggle it themselves in the admin.
+		 *
+		 * @since 1.0
+		 *
+		 * @param array $templates Supportable templates.
+		 */
+		$templates = apply_filters( 'amp_supportable_templates', $templates );
```

Comment:
This filter is already well-documented.

 But maybe it could also mention that each `$template` in `$templates` [can have](https://github.com/Automattic/amp-wp/pull/1235/files#diff-2585472f207548f626a43a7ff3cab922R386) an `$id` or `'callback'` that is a method of `WP_Query` (if I understand this right). Or it can simply be a `callable`.

Reply:
Thanks, this looks good.